struct.md

```diff
@@ -0,0 +1,310 @@
+
+
+```
+DeliveryList:
+  datatype: Types.DeliveryInfo[]
+  type: sensor
+  description: List of deliveries
+```
+
+By default the array has an arbitrary number of element and may be empty.
+If a fixed size array is wanted the keyword `arraysize` can be used to specify size:
+
+```
+DeliveryList:
+  datatype: Types.DeliveryInfo[]
+  arraysize: 5
+  type: sensor
+  description: List of deliveries
+```
+
+
+### Expectations on VSS implementations (e.g. VISS, KUKSA.val)
+
+For array types (like above) VSS implementations may support several mechanisms
+
+* It is expected that they can support read/write/subscribe of the whole array, i.e. write all or read all in the same request
+* They may optionally support additional operations like
+    * Writing/Reading a single instance, e.g. `DeliveryList[2]` (index mechanism is implementation dependent)
+    * Appending/Deleting individual instances
+    * Searching for instances with specific conditions.
+
+## Structure in Structure
+
+It shall be possible to refer to a structure type from within a structure
+
+```
+OpenHours:
+  type: struct
+  description: A struct type containing information on open hours
+  
+OpenHours.Open:
+  datatype: uint8
+  type: item
+  max: 24
+  description: Time the address opens
+  
+OpenHours.Close:
+  datatype: uint8
+  type: item
+  max: 24
+  description: Time the address close
+
+DeliveryInfo:
+  type: struct
+  description: A struct type containing info for each delivery
+  
+DeliveryInfo.Address:
+  datatype: string
+  type: item
+  description: Destination address
+
+DeliveryInfo.Receiver:
+  datatype: string
+  type: item
+  description: Name of receiver
+  
+DeliveryInfo.Open:
+  datatype: OpenHours
+  type: item
+  description: When is receiver available
+
+```
+
+### Order of declaration/definition
+
+The order of declaration/definition shall not matter.
+As signals and types are defined in different trees this is a topic only for struct definitions referring to other struct definitions.
+A hypothetical example is shown below. An item in the struct `DeliveryInfo` can refer to the struct `OpenHours` even if that struct
+is defined further down in the same file.
+
+```
+DeliveryInfo:
+  type: struct
+  description: A struct type containing info for each delivery
+
+...
+
+DeliveryInfo.Open:
+  datatype: OpenHours
+  type: item
+  description: When is receiver available
+
+OpenHours:
+  type: struct
+  description: A struct type containing information on open hours
+
+...
+
+```
+
+
+## Inline Struct
+
+Inline/anonymous structs shall not be supported!
+
+## Default Values
+
+VSS supports [default values for attributes](https://covesa.github.io/vehicle_signal_specification/rule_set/data_entry/attributes/),
+and there is a [discussion](https://github.com/COVESA/vehicle_signal_specification/issues/377)
+to allow it also for sensors/actuators. 
+
+It is proposed for now that default values shall not be supported for signals of struct type.
```

Comment:
Similar to allowed values, would we allow default values on the `item` nodes for supported data types?

Reply:
We could, but then we need to be clear on what it actually means. Will it just be a hint to implementations, (e.g. not necessary to explicitly set field C before sending the struct), or will there also be a bigger meaning - like if all items have a default value, then we also have a default value for the signal using that struct type?

Reply:
Adding text that default should be valid for items, and that if all items have default then the signal also implicitly has a default.